crates/turborepo-lib/src/task_graph/visitor.rs

```diff
@@ -71,16 +83,21 @@ impl<'a> Visitor<'a> {
     pub fn new(
         package_graph: Arc<PackageGraph>,
         run_cache: Arc<RunCache>,
+        run_tracker: RunTracker,
         opts: &'a Opts,
         package_inputs_hashes: PackageInputsHashes,
         env_at_execution_start: &'a EnvironmentVariableMap,
         global_hash: &'a str,
         global_env_mode: EnvMode,
         ui: UI,
         silent: bool,
+        version: &'static str,
         manager: ProcessManager,
         repo_root: &'a AbsoluteSystemPath,
         global_env: EnvironmentVariableMap,
+        packages: HashSet<WorkspaceName>,
+        started_at: DateTime<Local>,
+        global_hash_inputs: GlobalHashableInputs<'a>,
```

Comment:
I'm not sure I get why adding these into the `Visitor` is warranted.

The `RunTracker` constructor takes `version` and `started_at`, I think it's reasonable to hold onto these and use them for the `finish` call at that level. This avoids us accidentally passing in different values for the run start API call and the run finish API call.

`packages` and `global_hash_inputs` don't affect walking the task graph at all and can be provided in the call to `finish`

```suggestion
        manager: ProcessManager,
        repo_root: &'a AbsoluteSystemPath,
        global_env: EnvironmentVariableMap,
```

Reply:
Yeah I ended up a little too dogmatic in the refactoring. I wanted trackers to *only* have join handles and receivers, and the finish call to take no arguments. I think that's a good middle ground